petastorm/spark/spark_dataset_converter.py

```diff
@@ -594,18 +594,28 @@ def wait_for_file(path):
 
 def _check_dataset_file_median_size(url_list):
     fs, path_list = get_filesystem_and_path_or_paths(url_list)
+    SIZE_LIMIT_BYTES = 50 * 1024 * 1024
+    MB_BYTES = 1024 * 1024
 
     # TODO: also check file size for other file system.
     if isinstance(fs, LocalFileSystem):
         pool = ThreadPool(64)
         try:
             file_size_list = pool.map(os.path.getsize, path_list)
-            mid_index = len(file_size_list) // 2
-            median_size = sorted(file_size_list)[mid_index]
-            if median_size < 50 * 1024 * 1024:
-                logger.warning('The median size (%d) of these parquet files (%s) is too small.'
-                               'Increase file sizes by repartition or coalesce spark dataframe, which '
-                               'will help improve performance.', median_size, ','.join(url_list))
+            total_size = sum(file_size_list)
+            num_partitions = len(file_size_list)
+            if total_size > SIZE_LIMIT_BYTES:
```

Comment:
Shall we make the logic simpler:

~~~python
if len(path_lists) > 1:
 median_size = sorted(file_size_list, reverse=True)[mid_index]  # take the larger one if tie
 if median_size < RECOMMEND_FILE_SIZE_BYTES:
   logger.warning('...')
# no else branch
~~~